token/program-2022/src/extension/mod.rs

```diff
@@ -350,6 +350,39 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
             })
         }
     }
+    /// Unpack base state, leaving the extension data as a mutable slice
+    /// Checks the account_type, and initializes it if Uninitialized
+    ///
+    /// Fails if the base state is not initialized.
+    pub fn unpack_after_realloc(input: &'data mut [u8]) -> Result<Self, ProgramError> {
+        check_min_len_and_not_multisig(input, S::LEN)?;
+        let (base_data, rest) = input.split_at_mut(S::LEN);
+        let base = S::unpack(base_data)?;
+        if let Some((account_type_index, tlv_start_index)) = type_and_tlv_indices::<S>(rest)? {
+            // type_and_tlv_indices() checks that returned indexes are within range
+            let mut account_type = AccountType::try_from(rest[account_type_index])
+                .map_err(|_| ProgramError::InvalidAccountData)?;
+            if account_type == AccountType::Uninitialized {
+                rest[account_type_index] = S::ACCOUNT_TYPE.into();
+                account_type = S::ACCOUNT_TYPE;
+            }
```

Comment:
nit: we could probably create a common private function like `unpack_checked` or `unpacked_internal` which also takes in a flag to do the account type init, since it seems to be the only difference between `unpack` and `unpack_after_realloc`

Reply:
Oh, totally! I actually meant to go back and dedupe this way, but then forgot 😞
Thanks!